publish/src/coursier/publish/sonatype/SonatypeApi.scala

```diff
@@ -120,38 +122,57 @@ final case class SonatypeApi(
       isJson = true
     )(RawJson.codec)
 
+
   private def stagedRepoAction(
     action: String,
     profile: Profile,
     repositoryId: String,
-    description: String
-  ): Unit =
-    clientUtil.create(
-      s"${profile.uri}/$action",
-      post = Some(postBody(writeToArray(StagedRepositoryRequest(description, repositoryId)))),
-      isJson = true
-    )
+    description: String,
+    retries: Int
+  ): Unit = {
+    val url = s"${profile.uri}/$action"
+    val body = postBody(writeToArray(StagedRepositoryRequest(description, repositoryId)))
+    @tailrec
+    def sendRequest(attempt: Int): Unit = {
+      val resp = clientUtil.create(url, post = Some(body), isJson = true)
+
+      if (!resp.code.isSuccess) {
+        if (attempt >= retries)
+          throw new Exception(
+            s"Failed to get $url (http status: ${resp.code.code}, response: ${Try(new String(resp.body, StandardCharsets.UTF_8)).getOrElse("")})"
+          )
+
```

Comment:
Should we have something like `Thread.sleep(100)` here?

Reply:
Yeah, it makes sense to have one. If the Sonatype servers are flaky, better wait a bit before retrying. I think sbt-sonatype does similar things.